Update docs pages (#88)
ultralytics/hub-sdk: docs/model.md

---
comments: true
description: Learn how to manage machine learning models with Ultralytics HUB SDK operations.
keywords: Ultralytics HUB SDK, ML model management, machine learning model operations, HUB SDK Python
---

# Ultralytics HUB-SDK Model Management Operations Guide

Welcome to the Ultralytics HUB-SDK Model Management documentation! Whether you're just getting started with managing machine learning models or you're a seasoned data scientist looking for specific operation instructions, you've come to the right place. Let's embark on a smooth journey through the HUB-SDK features, ensuring you gain the know-how to efficiently manage your models.

## Retrieve a Model by its Unique Identifier

In machine learning workflows, you often need to access a specific model. With Ultralytics HUB-SDK, fetching a model by its ID is a breeze. This essential function sets up a model object based on the provided unique identifier, granting you full access to the model's details and operations.

```python
model = client.model("<Model ID>")
print(model.data)  # Outputs the model's metadata and configuration
```

## Access Project and Dataset Prerequisites

> Prior to model creation or training, ensuring the presence of both a project and a dataset is crucial. This straightforward code snippet helps you verify these components are available by initializing their objects. If either ID is missing, an exception is a clear indicator you need to address this before proceeding.

Reviewer: Little update with description. to add up it's not required. and not raise the exception.  
```suggestion
Prior to model creation or training, ensuring the presence of both a project and a dataset is crucial. This straightforward code snippet helps you verify these components are available by initializing their objects. While utilizing a project and dataset for organizing model training is beneficial, it's important to note that it is not mandatory. If either ID is missing, the object data (`project.data`, `dataset.data`) will  be empty.
```

project = client.project("<Project ID>")
dataset = client.dataset("<Dataset ID>")
```

## Create a New Model with Custom Configuration

Creating a new model tailored to your project requirements is made simple with this convenient function. Specify the model's name and associate it with your project and dataset. You can also customize configurations to fit your needs, such as setting the batch size or device, among others. Note that `projectId` and `datasetId` parameters are optional if you're not ready to tie the model to a project or dataset yet.

```python
data = {
    "meta": {"name": "sdk model"},  # Give your model a recognizable name
    "projectId": project.id,        # Associate with an existing project
    "datasetId": dataset.id,        # Associate with an existing dataset
    "config": {                     # Define hyperparameters and settings
        "batchSize": "-1",          
        "cache": "ram",
        "device": "name",
        "epochs": "5",
        "imageSize": "640",
        "patience": "5"             # Stop training if validation doesn't improve
    }
}
model = client.model()
model.create_model(data)  # Creates the model with your specified details
```

> ## Update an Existing Model's Metadata

Reviewer: ```suggestion
## Update an Existing Model's Metadata or Config
```

As projects develop, you might need to update a model's metadata, such as renaming it for clarity. The SDK provides a method to refresh these details effortlessly, minimizing manual errors and saving you precious time.

```python
model = client.model("<Model ID>")
model.update({"meta": {"name": "Updated Model Name"}})  # Renames the specified model
```

## Delete a Model Safely

Deleting a model is irreversible, so this function should be used with caution. When you're sure you want to remove a model from the system, the following command will permanently delete the specified model, along with all its associated data.

```python
model = client.model("Model ID")

> model.delete()  # Permanently deletes the specified model

Reviewer: ```suggestion
model.delete(hard=True)  # Permanently deletes the specified model
```

```

> 

Reviewer: ```suggestion
By default, the delete method performs a soft delete, marking the model as inactive without removing it permanently. If you want to perform a hard delete and remove the model along with its associated data permanently, you can pass the argument hard=True as shown in the example above. Exercise caution when using the hard delete option, as it is irreversible and results in the complete removal of the specified model from the system.

```
